superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -744,7 +746,10 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
       });
     }
 
-    setDB({ type: ActionType.addTableCatalogSheet });
+    if (database_name === 'Google Sheets') {
```

Comment:
Shouldn't this be checking for the engine/backend instead of the DB Name which might be changed by users?

Reply:
This is different than what the user names their db. But the engine would also work.

Reply:
i'll make a follow up PR with the enums of the engines so we can leverage that for checking vs. the string